### PR TITLE
Upgrade to nodeJS version 20

### PR DESCRIPTION
**Pull request by chmanie**
## Description

This PR updates nodeJS to version 20.11 everywhere (except for the network image with stays at 16.x).

Here are some notable changes:

- Use of node 20.11 in nvmrc, all workflows and all containers (except the network container)
- Replace `colony-event-metadata-parser` with the new `@colony/events`
- Update relevant dependencies (mostly our libraries, postcss and `eslint-plugin-jsdoc`)
- Adjustments and fixes for the on-demand environment
- Apply more `fetch` polyfills (in all lambdas in which colony-js is used)
- Move to `cross-fetch` in all lambdas

That's basically it. I recommend checking out this branch and running it locally. You can also run the on demand environment for this branch and check it out there.

### CODE REVIEW

File: postcss.config.js [hunk 1]
```diff
@@ -1,39 +1,21 @@
...
-      resolve(id) {
-        if (typeof aliases[id] == 'string') {
-          return path.resolve(__dirname, aliases[id]);
-        }
-        return id;
-      },
-    }),
-    postCSSPresetEnv({
-      features: {
-        'nesting-rules': true,
```
Comment (chmanie) [inline postcss.config.js:26]: Nesting i now done by the tailwind nesting plugin and the color-mod function was never used, so we could remove the entire thing

File: postcss.config.js [hunk 2]
```diff
@@ -1,39 +1,21 @@
...
-        'color-mod-function': { unresolved: 'warn' },
-      },
-      importFrom: './src/styles/variables.css',
```
Comment (chmanie) [inline postcss.config.js:29]: This wasn't really used, only `variables.global.css`

File: postcss.config.js [hunk 3]
```diff
@@ -1,39 +1,21 @@
...
 const postCSSImport = require('postcss-import');
-const postCSSPresetEnv = require('postcss-preset-env');
 const postCSSSvg = require('postcss-svg');
 const svgoPlugins = require('./webpack.base').svgoPlugins;
 const tailwindcss = require('tailwindcss');
+const nesting = require('tailwindcss/nesting');
 const autoprefixer = require('autoprefixer');
 
-// Aliases for CSS @import rules (relative to this file)
-const aliases = {
```
Comment (chmanie) [inline postcss.config.js:10]: Was never imported

**Comment by chmanie** [comment]
I cherry-picked Alex' commit from here https://github.com/JoinColony/colonyCDapp/pull/1716 to test it. Will remove it once #1716 is merged and this is rebased

**Review by ArmandoGraterol — DISMISSED** [review]
LGTM 👍 Ran the dev env from 0 along with the script to create data, also created some actions and all seems to be working as expected

**Review by jakubcolony — DISMISSED** [review]
Dev env works all fine, thanks for looking after the dependencies!

**Review by jakubcolony — APPROVED** [review]

**Review by bassgeta — APPROVED** [review]

**Review by ArmandoGraterol — APPROVED** [review]

**Pull request merged**